src/climate_learn/data/tasks/forecasting.py

```diff
@@ -86,7 +88,8 @@ def create_inp_out(self, index) -> Tuple[np.ndarray, np.ndarray]:
             idx = index + self.window * i
             inp.append(self.inp_data[idx])
         inp = np.stack(inp, axis=0)
-        out_idx = index + (self.history - 1) * self.window + self.pred_range
+
+        out_idx = index + (self.history - 1) * self.window + (self.pred_range // self.subsample)
```

Comment:
`self.window` should be divided by `self.subsample`. Also even though this problem for getting `out_idx` would be right after this change but it still won't solve the problem raised in the previous comment. Also calculation for `idx` would also need to change as you would need to divide `self.window` by `self.subsample` there.